server/grpcservice/service.go

```diff
@@ -269,6 +269,54 @@ func (s *Service) DropTable(ctx context.Context, req *metaservicepb.DropTableReq
 	}, nil
 }
 
+// RouteTables implements gRPC CeresmetaServer.
+func (s *Service) RouteTables(ctx context.Context, req *metaservicepb.RouteTablesRequest) (*metaservicepb.RouteTablesResponse, error) {
+	ceresmetaClient, err := s.getForwardedCeresmetaClient(ctx)
+	if err != nil {
+		return &metaservicepb.RouteTablesResponse{Header: responseHeader(err, "grpc routeTables")}, nil
+	}
+
+	// Forward request to the leader.
+	if ceresmetaClient != nil {
+		return ceresmetaClient.RouteTables(ctx, req)
+	}
+
+	tables, version, err := s.h.GetClusterManager().RouteTables(ctx, req.GetHeader().GetClusterName(), req.GetSchemaName(), req.GetTableNames())
+	if err != nil {
+		return &metaservicepb.RouteTablesResponse{Header: responseHeader(err, "grpc routeTables")}, nil
+	}
+
+	entries := make(map[string]*metaservicepb.RouteEntry, len(tables))
+	for tableName, entry := range tables {
+		nodeShards := make([]*metaservicepb.NodeShard, 0, len(entry.NodeShards))
+		for _, nodeShard := range entry.NodeShards {
+			nodeShards = append(nodeShards, &metaservicepb.NodeShard{
+				Endpoint: nodeShard.Endpoint,
+				ShardInfo: &metaservicepb.ShardInfo{
+					ShardId: nodeShard.ShardInfo.ShardID,
+					Role:    nodeShard.ShardInfo.ShardRole,
+				},
+			})
+		}
+
+		entries[tableName] = &metaservicepb.RouteEntry{
+			Table: &metaservicepb.TableInfo{
+				Id:         entry.Table.ID,
+				Name:       entry.Table.Name,
+				SchemaId:   entry.Table.SchemaID,
+				SchemaName: entry.Table.SchemaName,
+			},
+			NodeShards: nodeShards,
+		}
+	}
+
```

Comment:
Define a specific method for this conversion and keep the gRPC service method clean and simple.